lib/rules/quotes.js

```diff
@@ -228,6 +228,34 @@ module.exports = {
             }
         }
 
+        /**
+         * Checks whether or not a given TemplateLiteral node is using a template literal feature.
```

Comment:
it seems confusing to me. all  templateLiteral nodes are using template literal feature, no?

Reply:
For example, the TemplateLiteral node `` `hello world` `` is not using a template literal feature, whereas `` `hello ${name}` `` is using one. This matters because the 'quotes' rule may want to disallow an unnecessary template literal from being used, when a simple string literal `'hello world'` would suffice.

Reply:
Yeah, my example in the last comment is using a template expression. But expressions are just one feature of template literals. This function is checking if any of the three template literal features are in use.

Reply:
In that case I'm also confused -- what three features are you referring to?

Reply:
The features are:

1. Expression interpolation
2. Tagged templates
3. Multi-line strings (linebreaks)

The official documentation mentions these too: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Note that the existing code here was checking for these same features, but I'm trying to make these checks clearer by moving them into a helper function.

Reply:
Is there still confusion around this? Happy to make improvements.

Reply:
It might be a good idea to list the three features explicitly in the comment since it seems like the word "feature" has the potential to cause confusion. Aside from that, this change looks good to me.